modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/ghoul.dm

```diff
@@ -0,0 +1,282 @@
+//////////////////
+
+/datum/species/ghoul/disarm(mob/living/carbon/human/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
+	// Targeting Self? With "DISARM"
+	if (user == target)
+		var/target_zone = user.zone_selected
+		var/list/allowedList = list ( BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG )
+		var/obj/item/bodypart/affecting = user.get_bodypart(check_zone(user.zone_selected)) //stabbing yourself always hits the right target
+
+		if ((target_zone in allowedList) && affecting)
+
+			if (user.handcuffed)
+				to_chat(user, span_alert("You can't get a good enough grip with your hands bound."))
+				return FALSE
+
+			// Robot Arms Fail
+			if (affecting.status != BODYPART_ORGANIC)
+				to_chat(user, "That thing is on there good. It's not coming off with a gentle tug.")
+				return FALSE
+
+			// Pry it off...
+			user.visible_message("[user] grabs onto [p_their()] own [affecting.name] and pulls.", span_notice("You grab hold of your [affecting.name] and yank hard."))
+			if (!do_mob(user,target))
+				return TRUE
+
+			user.visible_message("[user]'s [affecting.name] comes right off in their hand.", span_notice("Your [affecting.name] pops right off."))
+			playsound(get_turf(user), 'sound/effects/meatslap.ogg', 40, 1) //ill change these sounds later
+
+			// Destroy Limb, Drop Meat, Pick Up
+			var/obj/item/I = affecting.drop_limb()
+			if (istype(I, /obj/item/food/meat/slab))
+				user.put_in_hands(I)
+
+			return TRUE
+	return ..()
+
+/datum/species/ghoul/proc/handle_limb_mashing()
+	SIGNAL_HANDLER
+
+/datum/species/ghoul/spec_attacked_by(obj/item/I, mob/living/user, obj/item/bodypart/affecting, mob/living/carbon/human/H, modifiers)
+	handle_limb_mashing()
+	// MEAT LIMBS: If our limb is missing, and we're using meat, stick it in!
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		return
+	if(H.stat < DEAD && !affecting && istype(I, /obj/item/food/meat/slab))
+		var/target_zone = user.zone_selected
+		var/list/limbs = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+
+		if((target_zone in limbs))
+			if(user == H)
+				user.visible_message("[user] begins mashing [I] into [H]'s torso.", span_notice("You begin mashing [I] into your torso."))
+			else
+				user.visible_message("[user] begins mashing [I] into [H]'s torso.", span_notice("You begin mashing [I] into [H]'s torso."))
+
+			// Leave Melee Chain (so deleting the meat doesn't throw an error) <--- aka, deleting the meat that called this very proc.
+			spawn(1)
+				if(do_mob(user,H))
+					// Attach the part!
+					var/obj/item/bodypart/newBP = H.newBodyPart(target_zone, FALSE)
+					H.visible_message("The meat sprouts digits and becomes [H]'s new [newBP.name]!", span_notice("The meat sprouts digits and becomes your new [newBP.name]!"))
+					newBP.attach_limb(H)
+					qdel(I)
+					playsound(get_turf(H), 'sound/effects/meatslap.ogg', 50, 1)
+
+			return TRUE // True CANCELS the sequence.
+
+	return ..() // TRUE FALSE
```

Comment:
Why is a ghoul using raw meat to grow/lose limbs

Reply:
its pretty much pure muscle, and its readily available throughout the station